modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/SearchControls.js

```diff
@@ -67,14 +67,16 @@ const SearchControls = ({
 				</ClayInput.Group>
 
 				{searchData &&
-					Object.keys(searchData).map((key) => (
-						<ClayInput
-							key={key}
-							name={key}
-							type="hidden"
-							value={searchData[key]}
-						/>
-					))}
+					Object.keys(searchData).map((key) =>
+						searchData[key].map((value) => (
```

Comment:
```suggestion
						searchData[key].map((value, index) => {
							const keyValue = `${key}-${index}`;

							return (
								<ClayInput
									key={keyValue}
									name={key}
									type="hidden"
									value={value}
								/>
							);
						})

```

Reply:
Just going to chime in here seeing as it is a chance to spread around some knowledge about React semantics.

One thing to note about ```${key}-${index}``` is that it is really no better than using just `index`.

I am guessing that you're appending `index` instead of just using `index` alone because of the conventional advice against using `index`. The reason that advice exists is that it doesn't help React optimize DOM reconciliation operations; what it really needs is something uniquely identifying about each element, so that you can do, for example things like moving the first item in a list to the end:

```js
// before:                      after:
<Thing key="a">a</Thing>        <Thing key="b">b</Thing>
<Thing key="b">b</Thing>        <Thing key="c">c</Thing>
<Thing key="c">c</Thing>        <Thing key="a">a</Thing>
```

Given these unique keys, if we render with a different order like that React has the option of reducing that to these two operations:

1. Remove first child with key `a` from parent.
2. Append new last child with key `a` to parent.

Compare that with what happens if you use integer keys:

```js
// before:                      after:
<Thing key="1">a</Thing>        <Thing key="1">b</Thing>
<Thing key="2">b</Thing>        <Thing key="2">c</Thing>
<Thing key="3">c</Thing>        <Thing key="3">a</Thing>
```

Because all those keys make a false claim about the "identity" of each element, React is most likely going to do three operations now instead of two:

1. Set inner text of item with key `1` to `b`.
2. Set inner text of item with key `2` to `c`.
3. Set inner text of item with key `3` to `a`.

Obviously not a problem with a small list, but you can see how this could be inefficient if `n` grows large.

In this case in `SeachControls.js`, given that the `key` is non-unique, adding `index` to it isn't going to make the synthetic key that you generate any more "identifying" that just using `index` alone. Does that make sense?

If you _really_ wanted keys that labeled things in an "identifying" way, you'd actually want keys like this:

- `key1-0`
- `key1-1`
- `key2-0`
- `key2-1`

(ie. `0` and `1` to identify first and second instance of each actual key).

Of course, that's all absurdly complicated for no tangible benefit in this case, so just going with `index` is actually probably about as good as it gets here.

Reply:
Thanks @wincent .

In any case for everyone subscribed to this PR, do you need anymore help for our side? (I don't have access to the original branch so I can't push any changes there)

Reply:
cc: @cgoncas

Reply:
Hey @julien, for these cases, just fork the branch and create a new PR so we don't have to wait for @cgoncas to make the change or force her to get back to this for such a small adjustment. 

I think in this case the benefits of moving this forward without disrupting them further greater surpass the drawbacks.

Reply:
Hey @julien !!!

I have given you access to my report, in case you need to push some changes :)

Thank you!!